packages/styled-components/src/constructors/createGlobalStyle.js

```diff
@@ -12,22 +12,19 @@ import type { Interpolation } from '../types';
 
 type GlobalStyleComponentPropsType = Object;
 
-// place our cache into shared context so it'll persist between HMRs
-if (IS_BROWSER) {
-  window.scCGSHMRCache = {};
-}
-
 export default function createGlobalStyle(
   strings: Array<string>,
   ...interpolations: Array<Interpolation>
 ) {
   const rules = css(strings, ...interpolations);
   const styledComponentId = `sc-global-${hasher(JSON.stringify(rules))}`;
   const globalStyle = new GlobalStyle(rules, styledComponentId);
+  let count = 0;
 
   function GlobalStyleComponent(props: GlobalStyleComponentPropsType) {
     const styleSheet = useStyleSheet();
     const theme = useContext(ThemeContext);
+    const instance = useMemo(() => String(++count), []);
```

Comment:
`useMemo` is a [hint, not a semantic guarantee](https://reactjs.org/docs/hooks-faq.html#how-to-create-expensive-objects-lazily). It shouldn't contain side effects because React could re-execute it during any render cycle even if the deps haven't changed.

To actually guarantee that this remains stable across renders, you can use a lazily-initialized ref:

```js
const instanceRef = useRef(null)
if (instanceRef.current === null) instanceRef.current = String(++count)
const instance = instanceRef.current
```

Reply:
Fair enough yeah. Probably more concurrent-mode safe too.